File: day1/swift/puzzle1.swift
#!/usr/bin/swift

import Foundation

let dataFilePath = "../day1_data.txt"

// Read file
let fileContents = try String(contentsOfFile: dataFilePath, encoding: String.Encoding.utf8)
let depthMeasurements = fileContents.split(whereSeparator: \.isNewline).map( { Int($0)! } ) // Force unwrapping not ideal

// Filter by looking at previous measurement (guard against first value, otherwise IndexOutOfRange error)
let deeperMeasurements = depthMeasurements.enumerated().filter( { ($0.offset != 0) ? $0.element > depthMeasurements[$0.offset - 1] : false } )
print(deeperMeasurements.count)

File: day1/python/puzzle1.py
#!/usr/bin/env python3

data_file_name = "../day1_data.txt"

with open(data_file_name, 'r') as file:
    depth_measurements = [int(line.rstrip()) for line in file]

deeper_count = 0
for i, measurement in enumerate(depth_measurements):
    if i < 1:
        continue

    if measurement > depth_measurements[i - 1]:
        deeper_count += 1

print(deeper_count)